assets/src/application/services/apollo/mutations/useMutationWithFeedback.ts

```diff
@@ -0,0 +1,87 @@
+import { useCallback, useRef } from 'react';
+import { MutationTuple, useMutation } from '@apollo/react-hooks';
+import { DocumentNode } from 'graphql';
+import { OperationVariables } from 'apollo-client';
+import { v4 as uuidv4 } from 'uuid';
+
+import { MutationType } from '@application/services/apollo/mutations';
+import { SystemNotificationsToaster } from '@application/services/toaster/types';
+
+interface MutationWithFeedbackArgs {
+	typeName: string; // e.g. "Datetime", "Ticket", "PriceType"
```

Comment:
To keep this hook in `/application` I set `typeName` as `string` to avoid pulling from domain.